doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+* Just open the Suggestions UI with all suggestions sources, and group them by
+  the source of the suggestions. 
+
+This should cover most of the basic use cases for suggestions. 
+
+#### Who owns this menu?
+
+There was some discussion of who should own the suggestions menu. The control
+itself? Or the app hosting the control?
+
+A main argument for hosting this UI in the control itself is that any consumer
+of the `TermControl` should be able to display the [shell-driven autocompletion]
+menu. And they should get the UI from us "for free". Consumers shouldn't need to
+reimplement it themselves. In this scenarioThis probably could be done without
+many changes
+* Instead of operating on `Command`s and actions from the terminal settings,
+  the control could just know that all the entries in the menu are "send
+  input" "actions".
+* The control could offer a method to manually invoke the Suggestions UI for a
+  list of {suggestion, name, description} objects.
+* The app layer could easily translate between sendInput actions and these
+  pseudo-actions.
+
+A big argument in favor of having the app layer host the control: Consider an
+app like Visual Studio. When they embed the control, they'll want to style the
+shell-completions UI in their own way. They already have their own intellisense
+menu, and their own UI paradigm.
+
+For now, we'll leave this as something that's owned by the app layer. When we
+get around to finalizing the [shell-driven autocompletion] design, we can
+iterate on ideas for supporting both consumers that want to use a pre-built
+suggestions control, or consumers who want to bring their own.
+
+## Tenets
+
+<table>
+
+<tr><td><strong>Compatibility</strong></td><td>
+
+This shouldn't break any existing flows. This is a general purpose UI element,
+to be extended in a variety of ways. Those customizations will all be opt-in by
+the user, so I'm not expecting any breaking compatibility changes here.
+
+</td></tr>
+
+<tr><td><strong>Accessibility</strong></td><td>
+
+The Suggestions UI was designed with the goal of making commandline shell
+suggestions _more_ accessible. As Carlos previously wrote:
+
+> Screen readers struggle with this because the entire menu is redrawn every time, making it harder to understand what exactly is "selected" (as the concept of selection in this instance is a shell-side concept represented by visual manipulation).
+> 
+> ...
+> 
+> _\[Shell driven suggestions\]_ can then be leveraged by Windows Terminal to create UI elements. Doing so leverages WinUI's accessible design. 
+
+This will allow the Terminal to provide more context-relevant information to
+screen readers. 
+
+</td></tr>
+
+<tr><td><strong>Sustainability</strong></td><td>
+
+No sustainability changes expected.
+
+</td></tr>
+
+<tr><td><strong>Localization</strong></td><td>
+
+The localization needs of the Suggestions UI will be effectively the same as the
+needs of the Command Palette.

```

Comment:
Plus the need to consider "source" names; as well, the command palette spec missed the need for dual-localization search (english + native language). Do we have similar concerns here?